src/platform.ts

```diff
@@ -54,197 +66,540 @@ export function getPlatformDetails(): IPlatformDetails {
+        this.sysNativeWinPSExeVal   = new Lazy(() => this.findSysNativeWinPS());
+    }
+
+    /**
+     * The stable PowerShell 6+ installation.
+     * May be null if the installation directory is not present.
+     */
+    private get pwshStable(): IPossiblePowerShellExe {
+        return this.stablePwshExeVal.value;
+    }
+
+    /**
+     * The preview PowerShell 6+ installation.
+     * May be null if the installation directory is not present.
+     */
+    private get pwshPreview(): IPossiblePowerShellExe {
+        return this.previewPwshExeVal.value;
+    }
+
+    /**
+     * The stable 32-bit PowerShell 6+ installation.
+     * May be null if the installation directory is not present
+     */
+    private get pwsh32Stable(): IPossiblePowerShellExe {
+        return this.stable32BitPwshExeVal.value;
+    }
+
+    /**
+     * The preview 32-bit PowerShell 6+ installation.
+     * May be null if the installation directory is not present.
+     */
+    private get pwsh32Preview(): IPossiblePowerShellExe {
+        return this.preview32BitPwshExeVal.value;
+    }
+
+    /**
+     * PowerShell 6+ installation from an MSIX (through the Windows store).
+     * May be null if the AppX package is not found.
+     */
+    private get pwshMsix(): IPossiblePowerShellExe {
+        return this.msixExeVal.value;
+    }
+
+    /**
+     * PowerShell 6+ stable Snap installation on Linux.
+     */
+    private get pwshSnapStable(): IPossiblePowerShellExe {
+        return this.stableSnapExeVal.value;
+    }
+
+    /**
+     * PowerShell 6+ preview Snap installation on Linux.
+     */
+    private get pwshSnapPreview(): IPossiblePowerShellExe {
+        return this.previewSnapExeVal.value;
+    }
+
+    /**
+     * PowerShell 6+ .NET Core global tool installation.
+     */
+    private get pwshDotnetGlobalTool(): IPossiblePowerShellExe {
+        return this.dotnetGlobalToolExeVal.value;
+    }
+
+    /**
+     * The Windows PowerShell installation under the %windir%\System32 folder.
+     * This always exists.
+     */
+    private get sys32WinPS(): IPossiblePowerShellExe {
+        return this.sys32WinPSExeVal.value;
+    }
+
+    /**
+     * The 32-bit Windows PowerShell installation when running in a 64-bit process.
+     */
+    private get sysWoW64WinPS(): IPossiblePowerShellExe {
+        return this.sysWow64WinPSExeVal.value;
+    }
+
+    /**
+     * The 64-bit Windows PowerShell installation when running
+     * in a 32-bit process on a 64-bit Windows.
+     */
+    private get sysnativeWinPS(): IPossiblePowerShellExe {
+        return this.sysNativeWinPSExeVal.value;
+    }
+
+    /**
+     * Iterates through PowerShell installations on the machine according
+     * to configuration passed in through the constructor.
+     * PowerShell items returned by this object are verified
+     * to exist on the filesystem.
+     */
+    public *enumeratePowerShellInstallations(): Iterable<IPowerShellExeDetails> {
```

Comment:
is it worth it to have a helper function called `getFirstAvailablePowerShellInstallation()` that abstracts away the need for a user to know what to do with Iterables?